x-pack/legacy/plugins/fleet/public/pages/policy_details/components/datasources_table.tsx

```diff
@@ -40,73 +44,88 @@ export const DatasourcesTable: React.FC<Props> = ({
       packageTitle,
       packageVersion,
       packageDescription,
+      policies: policies?.length || 0,
     })
   );
 
+  const columns: EuiInMemoryTableProps['columns'] = [
+    {
+      field: 'name',
+      name: i18n.translate('xpack.fleet.policyDetails.datasourcesTable.nameColumnTitle', {
+        defaultMessage: 'Name',
+      }),
+    },
+    {
+      field: 'packageTitle',
+      name: i18n.translate('xpack.fleet.policyDetails.datasourcesTable.packageNameColumnTitle', {
+        defaultMessage: 'Package',
+      }),
+    },
+    {
+      field: 'packageVersion',
+      name: i18n.translate('xpack.fleet.policyDetails.datasourcesTable.packageVersionColumnTitle', {
+        defaultMessage: 'Version',
+      }),
+    },
+    {
+      field: 'streams',
+      name: i18n.translate('xpack.fleet.policyDetails.datasourcesTable.streamsCountColumnTitle', {
+        defaultMessage: 'Streams',
+      }),
+    },
+    {
+      name: i18n.translate('xpack.fleet.policyDetails.datasourcesTable.actionsColumnTitle', {
+        defaultMessage: 'Actions',
+      }),
+      actions: [
+        {
+          render: ({ packageName, packageVersion }: any) => {
+            return (
+              <EuiLink
+                color="primary"
+                external
+                target="_blank"
+                href={`${window.location.origin}${framework.info.basePath}/app/epm#/detail/${packageName}-${packageVersion}`}
+              >
+                <FormattedMessage
+                  id="xpack.fleet.policyDetails.datasourcesTable.viewActionLinkText"
+                  defaultMessage="view"
+                />
+              </EuiLink>
+            );
+          },
+        },
+      ],
+      width: '100px',
+    },
+  ];
+
+  if (withPoliciesCount) {
+    columns.splice(columns.length - 1, 0, {
+      field: 'policies',
+      name: i18n.translate('xpack.fleet.policyDetails.datasourcesTable.policiesColumnTitle', {
+        defaultMessage: 'Policies',
+      }),
+      render: (policies: number) => {
+        return policies === 0 ? (
+          <EuiBadge>
+            <FormattedMessage
+              id="xpack.fleet.policyDetails.datasourcesTable.unassignedLabelText"
+              defaultMessage="Unassigned"
+            />
+          </EuiBadge>
+        ) : (
+          policies
```

Comment:
it's possible to display policies name in a tooltip?

Reply:
in general I'm not a big fan of displaying more information in UI via tooltips; for this instance I think linking to the policies table, filtered by data source, might be a better option. I'll create an issue to support filtering from query param